sdk/core/src/policies/retry_policies/retry_policy.rs

```diff
@@ -101,7 +101,7 @@ where
                         );
                         error
                     } else {
-                        log::error!("non-io error occurred which will not be retried: {}", error);
+                        log::debug!("non-io error occurred which will not be retried: {}", error);
```

Comment:
The change above makes a lot of sense. But the errors here are all non-HTTP errors. Wouldn't logging at an `error` level still make sense?